docs/hugo/content/design/secrets.md

```diff
@@ -0,0 +1,670 @@
+This is a [standard pattern](https://github.com/crossplane/crossplane-runtime/blob/6a7a44ac50aa1caca20a3cb5c215e6c03dc2b58e/pkg/resource/resource.go#L131)
+used across all of the providers.
+
+The destination of the `writeConnectionSecretToRef` is currently always a Kubernetes secret, but there is an [open issue](https://github.com/crossplane/crossplane/issues/2366) 
+requesting pluggable secret stores.
+
+**User provided secrets** seem to be 
+[automatically generated](https://github.com/crossplane/provider-azure/blob/f37af6dd4f9d7d10a14caf8270d38078aee06bbe/pkg/controller/database/mysqlserver/managed.go#L157)
+by Crossplane.
+
+### AWS Controller for Kubernetes (ACK)
+
+**User provided secrets** are provided via a [SecretKeyRef](https://github.com/aws-controllers-k8s/runtime/blob/main/apis/core/v1alpha1/secret.go). 
+This allows cross-namespace references to a secret. The specific `Key` of the secret is selected with a `Key string` parameter.
+
+It looks like there may [not currently be support](https://github.com/aws-controllers-k8s/community/issues/700) for key updates/rollover.
+
+**AWS generated secrets** I can't find any examples of. They do not seem to classify "endpoints" as a secret, as shown by
+[opensearchservice endpoint](https://aws-controllers-k8s.github.io/community/reference/opensearchservice/v1alpha1/domain/) and
+[cluster endpoint](https://aws-controllers-k8s.github.io/community/reference/eks/v1alpha1/cluster/).
+
+## Proposal
+
+### User specified secrets
+
+User specified secrets will be detected and transformed from `string` to a `SecretReference`:
+
+```go
+type SecretReference struct {
+    // Name is the name of the secret. The secret must be in the same namespace as the resource.
+    Name string
+
+    // Key is the key in the secret to use.
+    Key string
+}
+```
+
+Detection will be done with a combination of:
+1. Additions to ASO's configuration file to flag particular fields as secret (probably in the `ObjectModelConfiguration` section).
+2. Using the `"format": "password"` data from Swagger, such as that used by 
+   [MySQL Flexible Server](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2021-05-01/mysql.json#L1662). 
+   Note that not all specs have this, for example
+   [VMSS does not](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json#L9397).
+3. Using the `x-ms-secret` annotation, which I assume has the same meaning as `"format": "password"` although it's not actually exactly documented anywhere.
+
+This is a place where we can push changes upstream to flag things as passwords if they're not being flagged.
+
+#### Lifecycle
+Since these secrets were created _by the user_, the user owns the lifecycle of these secrets. They could be using the same secret across many resources, or 
```

Comment:
I think this should be present tense.

```suggestion
Since these secrets arecreated _by the user_, the user owns the lifecycle of these secrets. They could be using the same secret across many resources, or 
```